### DISCUSSION TITLE
Handling unexpected errors

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [NRWLDev/fastapi-problem#13] [Q&A] [answered]
I want to capture and log unexpected errors. I'm using the default handler which includes the exception handler: 
```
app = FastAPI()
add_exception_handler(
    app,
    logger=logger,
    )
```
The documentation at https://nrwldev.github.io/fastapi-problem/usage/ says:
_"To handle unexpected errors provide unhandled_wrappers, a dict mapping http status code to StatusProblem, the system key default is also accepted as the root wrapper for all unhandled exceptions."_

What does this mean wrt to the default handler?

**Comment by EdgyEdgemond** [comment]
Yeah, I need to make that section clearer. If you want to provide custom exceptions for unhandled errors, then you can provide that configuration.

All unexpected errors will get handled regardless of custom wrappers. Those allow you to provide a better 404 message for example.

A starlette raised `HTTPException(404)` will return like this.
```
        {
            "title": "Not Found",
            "details": "Not Found",
            "type": "http-not-found",
            "status": 404,
        }

````


If you provide:

```
class CustomNotFoundError(StatusProblem):
    status = 404
    title = "That endpoint can not be located."

app = FastAPI()
add_exception_handler(
    app,
    logger=logger,
    unhandled_wrappers={
        404: CustomNotFoundError,
    },
)
```

Would return the same `HTTPException(404)` from starlette like so:

```
        {
            "title": "That endpoint can not be located.",
            "details": "404: Not Found",
            "type": "custom-not-found",
            "status": 404,
        }
````

**Reply by EdgyEdgemond** [reply, marked as answer]
https://github.com/NRWLDev/fastapi-problem/tree/main/examples

Check out the recently added examples (namely builtin and custom, for hopefully more clarity on what `unhandled_wrappers` does.)

**Reply by dineshbvadhia** [reply]
Thank-you. The examples helped to clarify. My software is an api-based black box to run on remote computers. As such, the default add_exception_handler is best right now to capture and log unexpected errors.

Btw fastapi-problem is incredibly useful and of value :)
